lib/librato/metrics/persistence/direct.rb

```diff
@@ -18,9 +16,17 @@ def persist(client, queued, options={})
             requests = [queued]
           end
           requests.each do |request|
+            resource =
+              if queued[:tags] || queued[:multidimensional]
+                # request contains per-measurement tags
+                queued.delete(:multidimensional) if queued[:multidimensional]
```

Comment:
Scenario when top-level tags are not set, but per-measurement tags are. Instead of attempting to enumerate all measurements and checking each for per-measurement tags, set top-level `:multidimensional` flag and hit `/v1/measurements` if present. Worth noting this key is actually deleted before sending since `multidimensional` is an invalid param.